sphinxcontrib/katex-server.js

```diff
@@ -0,0 +1,142 @@
+const net = require("net");
+const process = require("process");
+
+let value = null;
+let katex_options = {};
+let socket = null;
+let port = null;
+// this is the downloaded katex javascript library.
+// need to keep it updated
+// path to the downloaded katex javascript library `katex.js`.
+let katex_path = "./katex";
+process.argv.forEach(function(arg) {
+    if (value == "katex_path") {
+        katex_path = arg;
+        value = null;
+    } else if (value == "socket_path") {
+        socket = arg;
+        value = null;
+    } else if (value == "socket_port") {
+        port = arg;
+        value = null;
+    } else {
+        if (arg == "--katex") {
+            value = "katex_path";
+        } else if (arg == "--socket") {
+            value = "socket_path";
+        } else if (arg == "--port") {
+            value = "socket_port";
+        } else {
+            // Ignore unknown/unexpected arguments, for example the path to this
+            // script
+        }
+    }
+});
+
+const katex = require(katex_path);
+
+let listen_options = {};
+if (socket !== null) {
+    listen_options["path"] = socket;
+} else if (socket_port !== null) {
```

Comment:
```suggestion
} else if (port !== null) {
```

will fix the error we are getting when building under Windows.

Reply:
I think it would be nice to add some tests for `KaTeXServer` and also add code coverage, but we or I can also work on this at a later stage.

Reply:
That will require some fixtures for this.